tests/Aspire.Hosting.Milvus.Tests/MilvusFunctionalTests.cs

```diff
@@ -0,0 +1,368 @@
+        {
+            [$"ConnectionStrings:{db.Resource.Name}"] = await db.Resource.ConnectionStringExpression.GetValueAsync(CancellationToken.None)
+        });
+
+        hb.AddMilvusClient(db.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        var milvusClient = host.Services.GetRequiredService<MilvusClient>();
+        await milvusClient.CreateDatabaseAsync("db1");
+        string collectionName = "book";
+        var collection = await milvusClient.CreateCollectionAsync(
+                collectionName,
+                new[] {
+                FieldSchema.Create<long>("book_id", isPrimaryKey:true),
+                FieldSchema.Create<long>("word_count"),
+                FieldSchema.CreateVarchar("book_name", 256),
+                FieldSchema.CreateFloatVector("book_intro", 2)
+                }
+            );
+
+        var collections = await milvusClient.ListCollectionsAsync();
+
+        Assert.Single(collections, c => c.Name == collectionName);
+    }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task WithDataVolumeShouldPersistStateBetweenUsages()
+    {
+        var builder1 = CreateDistributedApplicationBuilder();
+        builder1.Configuration["Parameters:apikey"] = MilvusToken;
+        var apiKey1 = builder1.AddParameter("apikey");
+        var milvus1 = builder1.AddMilvus("milvus", apiKey1);
+
+        // Use a deterministic volume name to prevent them from exhausting the machines if deletion fails
+        var volumeName = VolumeNameGenerator.CreateVolumeName(milvus1, nameof(WithDataVolumeShouldPersistStateBetweenUsages));
+        milvus1.WithDataVolume(volumeName);
+        string collectionName = "book";
+
+        using (var app = builder1.Build())
+        {
+            await app.StartAsync();
+
+            var hb = Host.CreateApplicationBuilder();
+
+            hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+            {
+                [$"ConnectionStrings:{milvus1.Resource.Name}"] = await milvus1.Resource.ConnectionStringExpression.GetValueAsync(CancellationToken.None)
+            });
+
+            hb.AddMilvusClient(milvus1.Resource.Name);
+
+            using (var host = hb.Build())
+            {
+                await host.StartAsync();
+
+                var milvusClient = host.Services.GetRequiredService<MilvusClient>();
+
+                var collection = await milvusClient.CreateCollectionAsync(
+                        collectionName,
+                        new[] {
+                FieldSchema.Create<long>("book_id", isPrimaryKey:true),
+                FieldSchema.Create<long>("word_count"),
+                FieldSchema.CreateVarchar("book_name", 256),
+                FieldSchema.CreateFloatVector("book_intro", 2)
+                        });
+            }
+
+            // Stops the container, or the Volume would still be in use
+            await app.StopAsync();
```

Comment:
This should be in a `finally` block like https://github.com/dotnet/aspire/pull/4910 does. And same for the other `StopAsync` in this test.